pyteal/ast/seq.py

```diff
@@ -47,13 +47,16 @@ def __init__(self, *exprs):
         if len(exprs) == 1 and isinstance(exprs[0], list):
             exprs = exprs[0]
 
+        filtered_exprs: list[Expr] = []
         for i, expr in enumerate(exprs):
             if not isinstance(expr, Expr):
-                raise TealInputError("{} is not a pyteal expression.".format(expr))
+                continue
```

Comment:
@barnjamin Optional:  If not opposed, can you add a brief inline comment explaining why `Seq` construction validation is lax?

The PR description captures the spirit of what I have in mind.  If you prefer as is, feel welcomed to resolve.